### DISCUSSION TITLE
The model doesn't match geojson

### DISCUSSION THREAD

**Discussion by YIYOU27** [visgl/deck.gl#8085] [Q&A]
I have a set of geojson, converted into shp by QGIS, and then make the shp into a model by blender. The size of the model is many, dozens of kilometers, and then import the model into deck.gl, and found that the model needs to be reduced to match the geojson, and the part of the model, such as the curved part, does not match the geojson. Moreover, when the view is zoomed in and out and dragged, the geojson and the model are offset. I am not sure which one is offset. Does anyone encounter this problem please? Help is much appreciated. (Machine translation, please forgive me if there is any disrespect)

**Comment by YIYOU27** [comment]
https://github.com/visgl/deck.gl/assets/45561106/ee0b5144-96aa-4abe-871f-1919681ba092

The line is geojson, and neither the model nor geojson has an elevation

**Comment by Pessimistress** [comment]
> I am not sure which one is offset

Maybe add a base map so you can see how well they match the context.

Assuming you are using `SimpleMeshLayer`/`ScenegraphLayer`, the vertex coordinates are interpreted as meter offsets. There is a precision limit to how accurate they can be mapped in the emulated Mercator projection system, see [here](https://github.com/uber-web/math.gl/blob/master/docs/developer-guide/geospatial/web-mercator-offset-accuracy.md#meter-offset-to-pixels). There are a few things you can try:

1. Make sure the geographical anchor of the layer, set by either `coordinateOrigin` or `getPosition`, is roughly centered in the model. The closer vertex positions are to 0, the more precision you will get.
2. Set the `_instanced: false` prop on the layer. This will improve the projection precision (from "unadjusted" to "adjusted" in the linked table).
3. If your model is very large (say >10km), the above steps may not fully solve your problem. You may need to split it into multiple smaller meshes and render them using separate layers.

Note these are just generic advices based on your description. You can get better help if you share your data and code.
